### ISSUE TITLE
"npm test" obliterates index.js (aka: document the Typescript voodoo)

### ISSUE THREAD

**Issue by stevage** [Turfjs/turf#1407]
I just tried making changes to `packages/turf-square-grid/index.js` and ran `npm run test`. The changes were instantly lost, replaced with a previous version of the file. By poking around, I think what's happening is that:

* `npm run test` first calls `tsc`
* `tsc` transpiles `index.ts` over the top of `index.js`.

Two problems here:

1. This is super unintuitive and very unfriendly to the new contributor.
2. None of the typescript stuff is documented anywhere. (Including the fact that `tsc` is a pre-requisite.)

I think it's very slowly dawning on me that Turf is now turning into a Typescript-first project? And instead of modifying index.js, I should have been modifying index.ts? Could I suggest that documenting this fact is kind of important?

There's no mention of this fact anywhere, just the cryptic remark "TypeScript is supported internally within each module, no installs required." in the README, which doesn't really mean anything to me. (And the last part is not true, none of the npm commands work if you don't `npm install -g tsc`).

**Comment by rowanwins** [comment]
Hi @stevage 

Sorry thought I dropped a note here on friday but obviously not. Yes you're correct that unfortunately at the moment things aren't well documented. 

A [decision](https://github.com/Turfjs/turf/pull/1240) was made to go all-in on typescript, unfortunately this implementation has taken longer than expected as one of the key contributors and drivers for the change has been tied up elsewhere.

I'm trying to pick up some pieces at the moment and understand where things are at, once I've got a good handle on things I'll relook at this.

Sorry for the confusion it's caused
